pkg/skaffold/runner/deployer.go

```diff
@@ -76,6 +77,13 @@ func GetDeployer(ctx context.Context, runCtx *runcontext.RunContext, labeller *l
 			if err != nil {
 				return nil, err
 			}
+			// Override the cluster on the runcontext.
+			// This is used to determine whether we should push images, and we want to avoid that unless explicitly asked for
+			runCtx.Cluster = config.Cluster{
```

Comment:
we explicitly disallow mixed mode with docker deployer and k8s deployer together, right? Then this setting won't affect any k8s deploys because there won't be any?

Reply:
that's right, there's a check at the end of this function which will error out. i'll add a comment to clarify